pkg/new-ui/v1beta1/backend.go

```diff
@@ -0,0 +1,359 @@
+package v1beta1
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+	"path/filepath"
+
+	"github.com/ghodss/yaml"
+	"google.golang.org/grpc"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	experimentv1beta1 "github.com/kubeflow/katib/pkg/apis/controller/experiments/v1beta1"
+	api_pb_v1beta1 "github.com/kubeflow/katib/pkg/apis/manager/v1beta1"
+	"github.com/kubeflow/katib/pkg/util/v1beta1/katibclient"
+)
+
+func NewKatibUIHandler(dbManagerAddr string) *KatibUIHandler {
+	kclient, err := katibclient.NewClient(client.Options{})
+	if err != nil {
+		log.Printf("NewClient for Katib failed: %v", err)
+		panic(err)
+	}
+	return &KatibUIHandler{
+		katibClient:   kclient,
+		dbManagerAddr: dbManagerAddr,
+	}
+}
+
+// ServeIndex will return index.html for any non-API URL
+func (k *KatibUIHandler) ServeIndex(buildDir string) func(w http.ResponseWriter, r *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		fp := filepath.Join(buildDir, "static/index.html")
+		log.Printf("Sending file %s for url: %s", fp, r.URL)
+
+		// never cache index.html
+		w.Header().Set("Cache-Control", "no-cache, no-store, must-revalidate, max-age=0")
+
+		// return the contents of index.html
+		http.ServeFile(w, r, fp)
+	}
+}
+
+func (k *KatibUIHandler) connectManager() (*grpc.ClientConn, api_pb_v1beta1.DBManagerClient) {
+	conn, err := grpc.Dial(k.dbManagerAddr, grpc.WithInsecure())
+	if err != nil {
+		log.Printf("Dial to GRPC failed: %v", err)
+		return nil, nil
+	}
+	c := api_pb_v1beta1.NewDBManagerClient(conn)
+	return conn, c
+}
+
+func (k *KatibUIHandler) SubmitYamlJob(w http.ResponseWriter, r *http.Request) {
+	//enableCors(&w)
+	var data map[string]interface{}
+
+	json.NewDecoder(r.Body).Decode(&data)
+
+	job := experimentv1beta1.Experiment{}
+	if yamlContent, ok := data["yaml"].(string); ok {
+		err := yaml.Unmarshal([]byte(yamlContent), &job)
+		if err != nil {
+			log.Printf("Unmarshal YAML content failed: %v", err)
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+		err = k.katibClient.CreateRuntimeObject(&job)
+		if err != nil {
+			log.Printf("CreateRuntimeObject from YAML failed: %v", err)
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+	}
+
+}
+
+func (k *KatibUIHandler) SubmitParamsJob(w http.ResponseWriter, r *http.Request) {
```

Comment:
We can remove this function.

Reply:
ACK, I'll remove it

Reply:
Resolving this convo, since we created the create_experiment route, as described in https://github.com/kubeflow/katib/pull/1427#discussion_r581081446